java/src/jni/h5util.c

```diff
@@ -1443,29 +1473,25 @@ h5str_dump_region_blocks_data(JNIEnv *env, h5str_t *str, hid_t region, hid_t reg
     hid_t    type_id   = H5I_INVALID_HID;
     int      ndims     = -1;
     int      ret_value = FAIL;
+    int      i;
+    char     tmp_str[256];
 
-    /*
-     * This function fails if the region does not have blocks.
-     */
-    H5E_BEGIN_TRY { nblocks = H5Sget_select_hyper_nblocks(region); }
-    H5E_END_TRY;
-
-    if (nblocks < 0)
-        H5_LIBRARY_ERROR(ENVONLY);
-
-    if ((ndims = H5Sget_simple_extent_ndims(region)) < 0)
+    if ((nblocks = H5Sget_select_hyper_nblocks(region_space)) <= 0) {
```

Comment:
This line should probably retain the H5E_BEGIN/END_TRY block so that a region with no blocks doesn't spit out an error message to the console. We'd need to make sure not to call the error handling code until after the END_TRY though.

Reply:
Okay, I can agree.